### PR TITLE
v7.22.1

### PR DESCRIPTION
**Pull request by romgrk**

### CODE REVIEW

File: CHANGELOG.md
Comment (arminmeh) [inline CHANGELOG.md:10]:
```suggestion
We'd like to offer a big thanks to the 5 contributors who made this release possible. Here are some highlights ✨:
```

File: CHANGELOG.md
Comment (LukasTy) [inline CHANGELOG.md:47]: I'm not sure we should include this in the changelog.
It's just a fix for the above (15177) PR.
```suggestion
```

File: packages/x-charts/package.json
Comment (LukasTy) [reply packages/x-charts/package.json:3]: Agreed, doesn't look like anything changed between `7.22.0` and now. 🤷

File: CHANGELOG.md
Comment (arminmeh) [inline CHANGELOG.md:68]:
```suggestion

### Docs
- [docs] Add section explaining how to keep the selection while filtering in Data grid docs (#15199) @arminmeh

```

File: CHANGELOG.md
Comment (KenanYusuf) [inline CHANGELOG.md:30]:
```suggestion
- [DataGrid] Prevent error when deleting the last row (#15153) @dpak-maurya
```

This should be attributed to the contributor, I just created the cherry-pick PR manually because I forgot to add the label before merging.

File: CHANGELOG.md
Comment (KenanYusuf) [inline CHANGELOG.md:17]:
```suggestion
@wojtkolos, @dpak-maurya.
```

File: packages/x-charts/package.json
Comment (romgrk) [reply packages/x-charts/package.json:3]: I'd rather do an empty release and keep all the version numbers in sync, wdyt?

File: CHANGELOG.md
Comment (KenanYusuf) [inline CHANGELOG.md:10]:
```suggestion
We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:
```

File: packages/x-charts/package.json
Comment (flaviendelangle) [reply packages/x-charts/package.json:3]: Feel like we are having this debate every two weeks :laughing: 
Our decision in the general meeting a few month ago was to just follow what Lerna updates, to at least be consistent everytime.

Comment (arminmeh) [reply packages/x-charts/package.json:3]: Last time when I was releasing we "agreed" to release all packages regardless
https://github.com/mui/mui-x/pull/14910#discussion_r1795107844
Script was updated to reflect that

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:56]: We are missing the "Same changes" for the tree view pro package :/

Comment (flaviendelangle) [inline CHANGELOG.md:56]:
```suggestion
- [TreeView] Export `TreeItem2DragAndDropOverlay` and `TreeItem2LabelInput` from the root of each package (#15208) @flaviendelangle 
- [TreeView] Fix drag and drop color usage (#15149) @LukasTy
```

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:19]:
```suggestion
@LukasTy, @arminmeh, @MBilalShafi, @KenanYusuf, @flaviendelangle.
```

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:10]:
```suggestion
We'd like to offer a big thanks to the 7 contributors who made this release possible. Here are some highlights ✨:
```

File: CHANGELOG.md
Comment (cherniavskii) [inline CHANGELOG.md:1]: Please sync with the latest master to include https://github.com/mui/mui-x/pull/15205 and https://github.com/mui/mui-x/pull/15155

Comment (romgrk) [reply CHANGELOG.md:1]: I have added those entries to the changelog.

File: packages/x-charts/package.json
Comment (LukasTy) [reply packages/x-charts/package.json:3]: Ok, the problem/question in this case is clear.
We have two cases:
1. We currently do not release when Lerna doesn't detect any changes and there were no specific changes in the package;
2. We release, but mention no changes in the CHANGELOG when no specific changes happened on the package, but Lerna detects changes (usually because of dependencies bumping)

This difference in approach is somewhat confusing and leads to unstable results. 🤔 

This is why I slightly prefer the more manual approach—releasing only the packages with changes to it and skipping in case of dependency bumping.

WDYT about the compromise of not listing the package in the changelog if it doesn't have any changes in the 2nd case? 🤔

Comment (flaviendelangle) [reply packages/x-charts/package.json:3]: > Last time when I was releasing we "agreed" to release all packages regardless

That's not what I understand from this discussion.
Lerna can release packages without PR tight to it (if you have a dep that was updated for example).

Comment (cherniavskii) [reply packages/x-charts/package.json:3]: > WDYT about the compromise of not listing the package in the changelog if it doesn't have any changes in the 2nd case? 🤔

I think we should list the package in the changelog if it was released, even if there are no notable changes.
And whether we should release the package or not – determined by Lerna.

Comment (LukasTy) [reply packages/x-charts/package.json:3]: Gotcha, I'm fine with that as well.
But this approach is still creating debates, that is why we can possibly improve it. 👍 🤔

File: CHANGELOG.md
Comment (cherniavskii) [reply CHANGELOG.md:1]: Thanks!
I just realized that I mentioned the `master` branch above – my bad, I meant the base branch (`v7.x`).

Could you sync the release branch with the base branch?
Most of the time this doesn't change anything, but this sync ensures that the code we aim to release passes the CI pipeline. In the past, it happened once (or more?) that after merging the release branch we couldn't release because the build was failing (a regression was added to the `master` branch in the meantime).

File: packages/x-charts/package.json
Comment (cherniavskii) [reply packages/x-charts/package.json:3]: >> Last time when I was releasing we "agreed" to release all packages regardless

> That's not what I understand from this discussion.

Here we go again 😅
I've opened https://github.com/mui/mui-x/pull/15219 to clarify this in the release instructions.
I'm gonna need each team's approval on it to make sure we're on the same page.
If there's anything to discuss or clarify further – leave a comment in https://github.com/mui/mui-x/pull/15219

**Review by cherniavskii — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

**Comment by Hsryj7** [comment]
-import { DataGrid } from '@mui/x-data-grid';
+import { DataGridPro } from '@mui/x-data-grid-pro';

**Review by Hsryj7 — APPROVED** [review]

File: CHANGELOG.md
Comment (Hsryj7) [reply CHANGELOG.md]: 3ea4248f1004dbb593d264ec2d0e5e20aad4f5c0

**Pull request merged**
